### ISSUE TITLE
How can I prevent Code from trying to reopen a corrupt file?

### ISSUE THREAD

**Issue by looeee** [microsoft/vscode#88263]
I've been stuck with this problem for a couple of weeks. 

I opened a file some time ago that crashed VSCode. This happened as a result of #86811.

Now whenever I open VSCode it tries to reopen the file and the window crashes. My only current solution is to open two windows at once and use the second window. 

So, the question is, how can I prevent code from trying to reopen this file?

Steps I have tried so far:

`code --disable-extensions` - same result
`code -n c:\` or `code -n` - opens two windows, one of which is the corrupt file.

Clear recent files from the file menu - no effect
Ctrl+shift+p - clear recently opened - no effect

Remove all files from the following folders:
C:\Users<user>\AppData\Roaming\Code\Cache
C:\Users<user>\AppData\Roaming\Code\CachedConfigurations
C:\Users<user>\AppData\Roaming\Code\CachedData
C:\Users<user>\AppData\Roaming\Code\CachedExtensions
C:\Users<user>\AppData\Roaming\Code\Code Cache
C:\Users<user>\AppData\Roaming\Code\GPUCache
C:\Users<user>\AppData\Roaming\Code\Local Storage

- no effect.

From dev tools - `window.localStorage.clear()` - no effect

Settings:
  "window.restoreWindows": "none", - no effect
  "files.hotExit": "off" - no effect

**Comment by connor4312** [comment]
Try removing `C:\Users\AppData\Roaming\Code\storage.json`

**Comment by looeee** [comment]
> Try removing %APPDATA%\Code\storage.json

Thanks for the suggestion. That also has no effect. 



![code](https://user-images.githubusercontent.com/5307958/72216609-c9a8da00-3555-11ea-944e-8463e001e062.jpg)

**Comment by looeee** [comment]
I found where the file is referenced: 

%APPDATA%\Code\Backups\1578350760097\file\0038ad32c7df2a82aa79855fc34adf9e - 

``` js
file:///d%3A/Downloads/icons/solid/user-alt-slash.svg {"mtime":1576150909759,"ctime":1576293291700,"size":467,"etag":"34amhr5n0f2","orphaned":false}
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><path fill="currentColor" d="M633.8 458.1L389.6 269.3C433.8 244.7 464 198.1 464 144 464 64.5 399.5 0 320 0c-67.1 0-123 46.1-139 108.2L45.5 3.4C38.5-2 28.5-.8 23 6.2L3.4 31.4c-5.4 7-4.2 17 2.8 22.4l588.4 454.7c7 5.4 17 4.2 22.5-2.8l19.6-25.3c5.4-6.8 4.1-16.9-2.9-22.3zM198.4 320C124.2 320 64 380.2 64 454.4v9.6c0 26.5 21.5 48 48 48h382.2L245.8 320h-47.4z"/></svg>
```

Deleting everything in the  %APPDATA%\Code\Backups folder seems to have fixed the issue. 

Any ideas what caused this?

**Comment by connor4312** [comment]
`hotExit` is the one that affects things. It looks like we only use that as a signal for future saving of state, not a signal to ignore any previously-saved state.

**Comment by looeee** [comment]
> hotExit is the one that affects things. 

As I mentioned above, setting `hotExit": "off"`  did not fix this. I had to search through all the files in `%APPDATA%\Code` to find where the corrupt file was a delete the references to fix this.

**Comment by connor4312** [comment]
What I mean to say is that's "as designed" 🙂 Maybe it should also actively ignore previous saves, cc @bpasero

**Comment by bpasero** [comment]
@looeee can you distill steps how to reproduce this? I would assume that when you disable hot exit you should get asked on shutdown to save or revert any dirty file (thus also the one that always opens). Once you do, the backup should disappear on disk.

**Comment by looeee** [comment]
> I would assume that when you disable hot exit you should get asked on shutdown to save or revert any dirty file

No, as I mentioned above setting hotExit": "off" or hotExit": "on" had no effect. 

In any case, I solved this as mentioned above by tracking down the file ref in %APPDATA%\Code\Backups and deleting everything from that folder. 

I cannot distil steps to reproduce it unfortunately, as far as I can tell the initial problem was caused by an extension but the problem persisted even when all extensions were disabled. 

For me, it would have saved me a lot of frustration if there was some command in code like "clear all cached files and backups".
